sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala

```diff
@@ -396,25 +396,40 @@ private object CosmosPartitionPlanner extends BasicLoggingTrait {
         Some(range)
       })
 
+    val result = new ArrayBuffer[PartitionMetadata]
     orderedRanges
-      .map(range => {
-        while (!SparkBridgeImplementationInternal.doRangesOverlap(range, startTokens(startTokensIndex)._1)) {
+      .foreach(range => {
```

Comment:
If there is one range we can not find any overlapping, what would happen? 

Previously, an exception is going to throw?

Reply:
It is fine to have one/multiple ranges without overlapping as long as there is at least one final range - can happen due to feed range filtering - but also multiple splits/merges between micro batch runs

Reply:
And if there is no range with overlap there is still an assert